### PR TITLE
LPS-166653 Allow same url title for different translations of friendly url

### PR DESCRIPTION
**Pull request by adolfopa**
# What is this about?

Allow to use the same url title for different translations for the same friendly url.

# How do I test this?

You can test it in 2 different places: journal and pages. For both, the steps a the following:
1. Create an entry with url title "abc" for the default language.
2. Edit the entry, add a url title translation for another language and put the same value ("abc").

This should *not* fail; both languages should have the same url title.

This should only work for the same entry; trying to reuse the url title for other entry should fail.

### CODE REVIEW

**Comment by liferay-lima** [comment]
ci:test:sf

**Comment by liferay-lima** [comment]
ci:test:relevant

File: modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java [hunk 1]
```diff
@@ -557,6 +564,17 @@ public void validate(
...
 		throws PortalException {
 
+		validate(
+			groupId, classNameId, classPK,
+			LocaleUtil.toLanguageId(LocaleUtil.getSiteDefault()), urlTitle);
+	}
+
+	@Override
+	public void validate(
+			long groupId, long classNameId, long classPK, String languageId,
```
Comment (AliciaGarciaGarcia) [inline modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java:574]: languageId never used

**Review by AliciaGarciaGarcia — CHANGES_REQUESTED** [review]

**Comment by WYZ-Issac** [comment]
ci:test:relevant

**Review by robertoDiaz — APPROVED** [review]

File: modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java (hunk 1 above)
Comment (robertoDiaz) [reply modules/apps/friendly-url/friendly-url-service/src/main/java/com/liferay/friendly/url/service/impl/FriendlyURLEntryLocalServiceImpl.java:574]: Agree

**Comment by robertoDiaz** [comment]
Some broken tests seem to be related with latest changes:

I.E: [com.liferay.journal.friendly.url.test.JournalArticleFriendlyURLTest.testUniqueFriendlyURLAfterUpdate](https://test-1-37.liferay.com/job/test-portal-acceptance-pullrequest-downstream(master)/318710//testReport/com.liferay.journal.friendly.url.test/JournalArticleFriendlyURLTest/testUniqueFriendlyURLAfterUpdate)

Seems to work in the origienal pull https://github.com/liferay-lima/liferay-portal/pull/3665

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:stop

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

**Comment by adolfopa** [comment]
ci:test:sf

**Comment by adolfopa** [comment]
ci:test:relevant

File: modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/friendly/url/test/JournalArticleFriendlyURLTest.java [hunk 2]
```diff
@@ -208,10 +208,10 @@ public void testUniqueFriendlyURLForSameTitlesWithExistingArticle()
...
 
 		Assert.assertEquals(
-			FriendlyURLNormalizerUtil.normalizeWithEncoding(title + "-2"),
+			FriendlyURLNormalizerUtil.normalizeWithEncoding(title + "-1"),
 			friendlyURLMap.get(LocaleUtil.US));
 		Assert.assertEquals(
-			FriendlyURLNormalizerUtil.normalizeWithEncoding(title + "-2-1"),
+			FriendlyURLNormalizerUtil.normalizeWithEncoding(title + "-1"),
 			friendlyURLMap.get(LocaleUtil.FRANCE));
 	}
```
Comment (robertoDiaz) [inline modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/friendly/url/test/JournalArticleFriendlyURLTest.java:216]: I'm not sure that we want to allow this behavior.

One of the conditions we agreed is that by now allow have the same url in multiple languajes is only aplicable to layouts. Anyway, if  @pabloagulla is ok with this, good to go.

Comment (robertoDiaz) [reply modules/apps/journal/journal-test/src/testIntegration/java/com/liferay/journal/friendly/url/test/JournalArticleFriendlyURLTest.java:216]: Of course, allowing that in all the cases is more consistent and simplifies the logic

**Comment by adolfopa** [comment]
ci:reevaluate:1307401_5216

**Comment by WYZ-Issac** [comment]
ci:test:relevant

**Comment by WYZ-Issac** [comment]
Hi @adolfopa, the failures are not related to this PR.

**Comment by adolfopa** [comment]
ci:forward

**Review by AliciaGarciaGarcia — APPROVED** [review]
ci:forward

**Comment by AliciaGarciaGarcia** [comment]
ci:forward

**Comment by adolfopa** [comment]
Forwarded to bchan: https://github.com/brianchandotcom/liferay-portal/pull/125786

**Pull request closed without merging**